src/Piling/PilingUpdater.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Autodesk.Revit.DB;
+using System;
+using System.Collections.Generic;
+using Autodesk.Revit.ApplicationServices;
+using Jpp.Cedar.Core;
+
+namespace Jpp.Cedar.Piling
+{
+    public class PilingUpdater : IUpdater
+    {
+        AddInId _appId;
+        UpdaterId _updaterId;
+
+        ISharedParameterManager _spManager;
+
+        bool registered = false;
+        Definition _eastingDefinition, _northingDefinition, _cutOffDefinition;
+
+        public PilingUpdater(AddInId id, ISharedParameterManager spManager)
+        {
+            if (id == null)
+                throw new System.ArgumentNullException(nameof(id));
+
+            if (spManager == null)
+                throw new System.ArgumentNullException(nameof(spManager));
+
+            _appId = id;
+            _updaterId = new UpdaterId(_appId, new Guid("ddb23f37-892e-4b43-9e8a-0ad8ff381b2b"));
+            _spManager = spManager;
```

Comment:
Consider single line join and heck.

_spManager = spManager ?? throw new System.ArgumentNullException(nameof(spManager));